### ISSUE TITLE
AddSelectionAbove and AddSelectionBelow remove selection instead of just adding

### ISSUE THREAD

**Issue by paolodelfino** [zed-industries/zed#17148]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

AddSelectionAbove and AddSelectionBelow remove the entire selection if you don't go in the same direction. I mean, let's say you first call AddSelectionAbove and then you also want to select below, therefore you AddSelectionbBelow, but this time it will not only add the selection below but also remove the selection above.

I find it a bug because it adds the extra thing of a cancellation that the name doesn't specify. If I want to revert by decision, in this case, I can for example press escape.

I usually use the correct behavior of this functions in vscode in cases like:
```c
#define MACRO\
do {\
    printf("Hello, World!\n");\
} while (0)
```
In this case I position myself to the longest line and add the selection above and below then put spaces and back delete and achieve something like:
```c
#define MACRO                   \
do {                                      \
    printf("Hello, World!\n"); \
} while (0)
```

The above is not possible in zed because of #17147 and #17142

### Environment

Zed: v0.152.0 (Zed Dev cdaa80fefbfd8e023f76849a99440aa4c52cfbc8)
OS: Windows 10.0.22631
Memory: 15.9 GiB
Architecture: x86_64
GPU: NVIDIA GeForce RTX 2060 || NVIDIA || 560.94

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by notpeter** [comment]
Can you do me a favor and edit the description to include concise step-by-step reproduction steps, consolidate #17142 #17147 and #17148 into a single issue and close the others?

I think you have a valid issue but I'm having trouble keeping track and I'd like discussion to occur in a single thread.  Thanks.

**Comment by paolodelfino** [comment]
What should the title be?


> Can you do me a favor and edit the description to include concise step-by-step reproduction steps, consolidate #17142 #17147 and #17148 into a single issue and close the others?
> 
> I think you have a valid issue but I'm having trouble keeping track and I'd like discussion to occur in a single thread. Thanks.

**Comment by daniel-pg** [comment]
Yes! I find personally this behavior extremely annoying because it forces me to move the cursor either to the very top or to the very bottom of the code block that I want to select first. This assumes that I have already paused to ponder where I want to start the selection, so if I change my mind and decide to select one more line in the opposite direction I have to start from scratch.

On the other hand, if I changed my mind and decided to unselect a line, I'd have to start from scratch, but this is rarely the case for me because I prefer to start selecting everything that I'm 100% sure I'll change first, and then I pause to think if I want to select the next lines that I'm unsure, so I find this much more comfortable cognitively.

Ultimately this is matter of personal preference and should be a configurable option.
